r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+
+### Creating your first project
+
+There are many ways one could organise a project folder. We'll set up a project
+directory folder using the
+[prodigenr](http://prodigenr.lukewjohnston.com) package:
+
+```r
+# prodigenr::setup_project("ProjectName")
+prodigenr::setup_project("learningr")
+```
+
+TODO: Use prodigenr or something else?
+
+When we use the `::` colon here, we are telling R "from the prodigenr package use
+the setup_project function". This function will then create the following
+folders and files:
+
+```
+learningr
+├── R
+│   ├── README.md
+│   ├── fetch_data.R
+│   └── setup.R
+├── data
+│   └── README.md
+├── doc
+│   └── README.md
+├── .gitignore
+├── DESCRIPTION
+├── learningr.Rproj
+└── README.md
+```
+
+This forces a specific, and consistent, folder structure to all your work. Think 
+of this like the "introduction", "methods", "results", and "discussion" sections 
+of your paper. Each project is then like a single manuscript or report, that
+contains everything relevant to that specific project. There is a lot of
+power in something as simple as a consistent structure. Projects are used to
+make life easier. Once a project is opened within RStudio the following actions
+are taken:
+
+- A new R session (process) is started.
+- The current working directory is set to the project directory.
+- RStudio project options are loaded.
+
+The README in each folder explains a bit about what should be placed there. But
+briefly:
+
+1. Documents like manuscripts, abstracts, and exploration type documents should 
+be put in the `doc/` directory (including [R Markdown][r-markdown] files which
+we will cover [later](#r-markdown)).
+1. Data, raw data, and metadata should be in either the `data/` directory or
+in `data-raw/` for the raw data.
+1. All R files and code should be in the `R/` directory.
+1. Name all new files to reflect their content or function. Follow the tidyverse 
+[style guide for file naming][r-style-guide-files].
+
+Note the `DESCRIPTION` file. This is used as metadata about the project and is 
+useful when working on R projects. For any project, it is **highly recommended**
+to use [version control]. We'll be covering version control in more detail
+[later](#version-control) in the course.
+
+TODO: Make sure the internal reference links are correct.
+
+### Exercise: Better file naming
+
+Look at the list of file names below. Which file names are good names and which
+shouldn't you use? 
+
+```
+fit models.R
+fit-models.R
+foo.r
+stuff.r
+get_data.R
+Manuscript version 10.docx
+manuscript.docx
+new version of analysis.R
+trying.something.here.R
+plotting-regression.R
+utility_functions.R
+code.R
+```
+
+### Should you keep your data under version control?
+
+We have a `data/` folder for a reason. But you might not want to keep the data
+under version control, for several reasons:
+
+1. It's a large dataset (tens or more of Mb file size)
+2. There are sensitive or personal information in the data
```

Comment:
"or personal" is a special case of "sensitive" ?

Reply:
well, based on the new GDPR rules in EU, personal doesn't necessarily have to be sensitive... if enough data is public that could someone could "theoretically" identify the individual, you can't make it public. Even those the specific data variables aren't sensitive. If that makes sense. I revised it to be clearer though.